src/EntryPoint.sol

```diff
@@ -411,8 +461,14 @@ contract EntryPoint is
         } else {
             (gUsed, err) = _execute(encodedUserOp, combinedGasOverride);
         }
-        // Revert with `abi.encodePacked(bytes4(0xffffffff), abi.encode(gUsed, err))`.
         assembly ("memory-safe") {
+            let noRevertCaller := calldataload(0x24)
+            if noRevertCaller {
+                // Require that the balance of the no revert caller is `type(uint256).max`.
+                if add(balance(noRevertCaller), 1) { invalid() }
+                return(0x00, 0x00) //
+            }
```

Comment:
Note: `type(uint256).max + 1 == 0` (due to overflow).

Reply:
From a Swiss cheese perspective, I think checking that the caller has a balance of `type(uint256).max` is safer than requiring `msg.sender == address(this)` for the non-reverting workflow. 

It is impossible for any account to have a balance of `type(uint256).max` in practice. 

But a coding mistake might accidentally enable the EntryPoint call this simulate self-call function outside of simulation.